src/app/Mappings/components/MappingBuilder/MappingBuilder.tsx

```diff
@@ -42,19 +46,18 @@ export const MappingBuilder: React.FunctionComponent<IMappingBuilderProps> = ({
   };
 
   let instructionText = '';
-  let selectSourcePlaceholder = '';
-  let selectTargetPlaceholder = '';
+  const selectSourcePlaceholder = 'Select source...';
+  const selectTargetPlaceholder = 'Select target...';
+
   if (mappingType === MappingType.Network) {
     instructionText = 'Map source and target networks.';
-    selectSourcePlaceholder = 'Select source...';
-    selectTargetPlaceholder = 'Select target...';
   }
   if (mappingType === MappingType.Storage) {
     instructionText = 'Map source datastores to target storage classes.';
-    selectSourcePlaceholder = 'Select source...';
-    selectTargetPlaceholder = 'Select target...';
   }
 
+  const allSourcesExhausted = () => builderItems.length === availableSources.length;
```

Comment:
Same here, I don't think it needs to be a function, just a plain boolean.

Reply:
Ok that one is called only once although I thought this was clearer. I changed it.